controllers/node/controllers/gpu_controller.go

```diff
@@ -277,5 +281,25 @@ func (r *GpuReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				return ok
 			},
 		})).
+		Watches(&source.Kind{Type: &corev1.Node{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(predicate.Funcs{
+			CreateFunc: func(event event.CreateEvent) bool {
```

Comment:
xxxFunc := func(event)bool {
 _, ok1 := event.Object.(*corev1.Node).Labels[NvidiaGPUMemory]
				_, ok2 := event.Object.(*corev1.Node).Labels[NvidiaGPUProduct]
				_, ok3 := event.Object.(*corev1.Node).Status.Allocatable[NvidiaGPU]
}

CreateFunc: xxxFunc
DeleteFunc: xxxFunc

same func, not repeatly

Reply:
Because the parameters of Create Func and Delete Func are different, it is not easy to abstract into a method?

Reply:
> Because the parameters of Create Func and Delete Func are different, it is not easy to abstract into a method?

With the same implementation, it can definitely be extracted：
```
xxxFunc := func(obj Object)bool {
_, ok1 := obj.(*corev1.Node).Labels[NvidiaGPUMemory]
_, ok2 := obj.(*corev1.Node).Labels[NvidiaGPUProduct]
_, ok3 := obj.(*corev1.Node).Status.Allocatable[NvidiaGPU]
}
CreateFunc: func(event xxEvent) bool { xxxFunc(event)}
DeleteFunc: func(event xxEvent) bool { xxxFunc(event)}
```